transportable-udfs-trino/src/main/java/com/linkedin/transport/trino/TrinoFactory.java

```diff
@@ -137,25 +151,22 @@ public StdStruct createStruct(StdType stdType) {
   }
 
   @Override
-  public StdType createStdType(String typeSignature) {
-    if (metadata != null) {
-      return TrinoWrapper.createStdType(metadata.getType(applyBoundVariables(
-          parseTypeSignature(quoteReservedKeywords(typeSignature), ImmutableSet.of()),
-          functionBinding)));
+  public StdType createStdType(String typeSignatureStr) {
+    TypeSignature typeSignature = applyBoundVariables(parseTypeSignature(quoteReservedKeywords(typeSignatureStr), ImmutableSet.of()), functionBinding);
+    if (typeManager != null) {
+      return TrinoWrapper.createStdType(typeManager.getType(typeSignature));
     }
-    return TrinoWrapper.createStdType(functionDependencies.getType(
-        applyBoundVariables(parseTypeSignature(quoteReservedKeywords(typeSignature), ImmutableSet.of()),
-            functionBinding)));
+    return TrinoWrapper.createStdType(functionDependencies.getType(typeSignature));
   }
 
   public MethodHandle getOperatorHandle(
       OperatorType operatorType,
       List<Type> argumentTypes,
       InvocationConvention invocationConvention) throws OperatorNotFoundException {
-    if (metadata != null) {
-      return metadata.getScalarFunctionInvoker(metadata.resolveOperator(operatorType, argumentTypes),
+    if (functionManager != null) {
```

Comment:
in addition to the new check, should we also retain the original null check for metadata since we are using `metadata.resolveOperator(...)` ?